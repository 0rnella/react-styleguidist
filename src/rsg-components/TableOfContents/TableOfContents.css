.root {
	composes: reset font from "../../styles/common.css";
	list-style-type: none;
	padding: 15px;
}

.item {
	margin: 0 0 7px;
	padding: 0;
	font-size: 15px;
}

.link {
	composes: link from "../../styles/colors.css";
}
