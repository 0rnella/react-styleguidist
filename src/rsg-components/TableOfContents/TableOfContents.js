import React, { Component, PropTypes } from 'react';

import s from './TableOfContents.css';

class TableOfContents extends Component {

	constructor(props) {
		super(props);

		this.state = {
			searchTerm: '',
			filteredComponents:[]
		};

		this.handleChange = this.handleChange.bind(this);
	}

	searchFilter(value, component) {
		const regex = new RegExp(value.toLowerCase().toString(), 'g');
		return component.name.toLowerCase().match(regex);
	}

	handleChange(e) {

		let components = this.props.components;

		if(e.target.value !== '') {
			components = components.filter(this.searchFilter.bind(null, e.target.value));
		}

		this.setState({
			searchTerm: e.target.value,
			filteredComponents: components
		})
	}

	componentDidMount() {
		this.setState({
			filteredComponents: this.props.components
		 })
	}

	render() {
		const { filteredComponents, searchTerm } = this.state;
		return (
			<div>
				<input
					className={s.find}
					placeholder="Find component..."
					onChange={this.handleChange}
					value={searchTerm} />
				<ul className={s.root}>
					{filteredComponents.map(({name}) => (
						<li className={s.item} key={name}>
							<a className={s.link} href={'#' + name}>{name}</a>
						</li>
					))}
				</ul>
			</div>
		);
	}
};

TableOfContents.propTypes = {
	components: PropTypes.array.isRequired
};

export default TableOfContents;
