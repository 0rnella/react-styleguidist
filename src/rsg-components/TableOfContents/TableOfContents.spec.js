import React from 'react';
import TableOfContents from './TableOfContents';

const components = [
	{
		name: 'Button',
	},
	{
		name: 'Input',
	},
	{
		name: 'Textarea',
	},
];

const sections = [
	{
		name: 'Introduction',
		content: 'intro.md',
	},
	{
		name: 'Buttons',
		components: [
			{
				name: 'Button',
			},
		],
	},
	{
		name: 'Forms',
		components: [
			{
				name: 'Input',
			},
			{
				name: 'Textarea',
			},
		],
	},
];

it('should render a renderer', () => {
	const actual = shallow(
		<TableOfContents components={components} sections={[]} />
	);

	expect(shallowToJson(actual)).toMatchSnapshot();
});

it('should render renderer with sections with nested components', () => {
	const actual = shallow(
		<TableOfContents components={[]} sections={sections} />
	);

	expect(shallowToJson(actual)).toMatchSnapshot();
});


it('should filter list when search field contains a query', () => {
	const searchTerm = 'but';
	const actual = shallow(
		<TableOfContents components={components} sections={[]} />
	);

	expect(shallowToJson(actual)).toMatchSnapshot();

	actual.setState({ searchTerm });

	expect(shallowToJson(actual)).toMatchSnapshot();
});

it('should render a filtered list, should hide empty sections', () => {
	const searchTerm = 'inp';
	const actual = shallow(
		<TableOfContents components={[]} sections={sections} />
	);

	expect(shallowToJson(actual)).toMatchSnapshot();

	actual.setState({ searchTerm });

	expect(shallowToJson(actual)).toMatchSnapshot();
});

it('should filter section names', () => {
	const searchTerm = 'frm';
	const actual = shallow(
		<TableOfContents components={[]} sections={sections} />
	);

	expect(shallowToJson(actual)).toMatchSnapshot();

	actual.setState({ searchTerm });

	expect(shallowToJson(actual)).toMatchSnapshot();
});
