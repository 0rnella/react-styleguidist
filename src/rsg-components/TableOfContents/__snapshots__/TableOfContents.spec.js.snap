exports[`test renderer should render table of contents 1`] = `
<div>
  <div>
    <div>
      <input
        onChange={[Function]}
        placeholder="Filter by name"
        value="foo" />
    </div>
    <div>
      foo
    </div>
  </div>
</div>
`;

exports[`test should filter list when search field contains a query 1`] = `
<_class
  items={
    <_class
      items={
        Array [
          Object {
            "name": "Button",
          },
          Object {
            "name": "Input",
          },
          Object {
            "name": "Textarea",
          },
        ]
      } />
  }
  onSearchTermChange={[Function]}
  searchTerm="" />
`;

exports[`test should filter list when search field contains a query 2`] = `
<_class
  items={
    <_class
      items={
        Array [
          Object {
            "name": "Button",
          },
        ]
      } />
  }
  onSearchTermChange={[Function]}
  searchTerm="but" />
`;

exports[`test should filter section names 1`] = `
<_class
  items={
    <_class
      items={
        Array [
          Object {
            "content": <_class
              items={Array []} />,
            "heading": true,
            "name": "Introduction",
          },
          Object {
            "content": <_class
              items={
                    Array [
                          Object {
                            "name": "Button",
                          },
                        ]
              } />,
            "heading": true,
            "name": "Buttons",
          },
          Object {
            "content": <_class
              items={
                    Array [
                          Object {
                            "name": "Input",
                          },
                          Object {
                            "name": "Textarea",
                          },
                        ]
              } />,
            "heading": true,
            "name": "Forms",
          },
        ]
      } />
  }
  onSearchTermChange={[Function]}
  searchTerm="" />
`;

exports[`test should filter section names 2`] = `
<_class
  items={
    <_class
      items={
        Array [
          Object {
            "content": <_class
              items={Array []} />,
            "heading": true,
            "name": "Forms",
          },
        ]
      } />
  }
  onSearchTermChange={[Function]}
  searchTerm="frm" />
`;

exports[`test should render a filtered list, should hide empty sections 1`] = `
<_class
  items={
    <_class
      items={
        Array [
          Object {
            "content": <_class
              items={Array []} />,
            "heading": true,
            "name": "Introduction",
          },
          Object {
            "content": <_class
              items={
                    Array [
                          Object {
                            "name": "Button",
                          },
                        ]
              } />,
            "heading": true,
            "name": "Buttons",
          },
          Object {
            "content": <_class
              items={
                    Array [
                          Object {
                            "name": "Input",
                          },
                          Object {
                            "name": "Textarea",
                          },
                        ]
              } />,
            "heading": true,
            "name": "Forms",
          },
        ]
      } />
  }
  onSearchTermChange={[Function]}
  searchTerm="" />
`;

exports[`test should render a filtered list, should hide empty sections 2`] = `
<_class
  items={
    <_class
      items={
        Array [
          Object {
            "content": <_class
              items={
                    Array [
                          Object {
                            "name": "Input",
                          },
                        ]
              } />,
            "heading": true,
            "name": "Forms",
          },
        ]
      } />
  }
  onSearchTermChange={[Function]}
  searchTerm="inp" />
`;

exports[`test should render a renderer 1`] = `
<_class
  items={
    <_class
      items={
        Array [
          Object {
            "name": "Button",
          },
          Object {
            "name": "Input",
          },
          Object {
            "name": "Textarea",
          },
        ]
      } />
  }
  onSearchTermChange={[Function]}
  searchTerm="" />
`;

exports[`test should render renderer with sections with nested components 1`] = `
<_class
  items={
    <_class
      items={
        Array [
          Object {
            "content": <_class
              items={Array []} />,
            "heading": true,
            "name": "Introduction",
          },
          Object {
            "content": <_class
              items={
                    Array [
                          Object {
                            "name": "Button",
                          },
                        ]
              } />,
            "heading": true,
            "name": "Buttons",
          },
          Object {
            "content": <_class
              items={
                    Array [
                          Object {
                            "name": "Input",
                          },
                          Object {
                            "name": "Textarea",
                          },
                        ]
              } />,
            "heading": true,
            "name": "Forms",
          },
        ]
      } />
  }
  onSearchTermChange={[Function]}
  searchTerm="" />
`;
